src/libcollections/btree/map.rs

```diff
@@ -875,20 +874,30 @@ impl<K: Ord, V: Ord> Ord for BTreeMap<K, V> {
     }
 }
 
-#[stable]
-impl<K: Show, V: Show> Show for BTreeMap<K, V> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        try!(write!(f, "BTreeMap {{"));
-
-        for (i, (k, v)) in self.iter().enumerate() {
-            if i != 0 { try!(write!(f, ", ")); }
-            try!(write!(f, "{:?}: {:?}", *k, *v));
-        }
+macro_rules! fmt_btree_map {
+    ($($Trait:ident),*) => {
+        $(
+            impl<K: fmt::$Trait, V: fmt::$Trait> fmt::$Trait for BTreeMap<K, V> {
+                fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                    try!(write!(f, "BTreeMap {{"));
+
+                    for (i, (k, v)) in self.iter().enumerate() {
+                        if i != 0 { try!(write!(f, ", ")); }
+                        try!(fmt::$Trait::fmt(k, f));
```

Comment:
This isn't quite right as it doesn't handle the formatting flags in the `Formatter` just right. This will forward any requested padding to the underlying implementation which isn't necessarily desired. You may want to pass around `fmt::Arguments` using the `format_args!` macro instead.